File: src/main/java/cat/xlagunas/andrtc/user/UserRepository.java
package cat.xlagunas.andrtc.user;

import cat.xlagunas.andrtc.user.ExistingUserException;
import cat.xlagunas.andrtc.user.UserNotFoundException;
import cat.xlagunas.andrtc.user.User;

import java.util.List;
import java.util.Optional;

public interface UserRepository {

    long insertUser(User user) throws ExistingUserException;

    User findUser(long userId) throws UserNotFoundException;

    User findUser(String username) throws UserNotFoundException;

    Optional<User> findUserOptional(long userId);

    List<User> findUsers(String username);

    boolean updatePassword(long userId, String password);

    boolean updateProfilePic(long userId, String thumbnail);
}

File: src/main/java/cat/xlagunas/andrtc/user/UserRepository.kt
package cat.xlagunas.andrtc.user

import java.util.Optional

interface UserRepository {

    @Throws(ExistingUserException::class)
    fun insertUser(user: User): Long

    @Throws(UserNotFoundException::class)
    fun findUser(userId: Long): User

    @Throws(UserNotFoundException::class)
    fun findUser(username: String): User

    fun findUserOptional(userId: Long): Optional<User>

    fun findUsers(username: String): List<User>

    fun updatePassword(userId: Long, password: String): Boolean

    fun updateProfilePic(userId: Long, thumbnail: String): Boolean
}
